### ISSUE TITLE
add an additional transport layer

### ISSUE THREAD

**Issue by benbrittain** [webRTC-io/webRTC.io#6]
possibly XMLHttpRequest? In the future we are going to want webRTC as a transport layer, but we need a data channel first.

**Comment by auchenberg** [comment]
Why not just use socket.io as the transport-layer abstraction?

**Comment by VGubarev** [comment]
Because if your browser supports WebRTC, it just supports WebSocket. And WS it's better data- chanel than long pooling, ajax etc.

**Comment by auchenberg** [comment]
I agree that WS is a better data-channel, but the reality is that most server-setup's don't allow WS due to firewall and loadbalancer issues. This is the main reason why I see a need to an alternative transport method for the data-channel. For example isn't not possible to deploy WebRTC.io on PAAS platforms like Heroku, AppFog, Nodester, etc, because they don't support WS (yet).

Socket.io would be a drop-in replacement for the WS-server, since it can be used as a cross-browser WS implementation. 

The WebRTC P2P connection between is a different thing, since the environment is much different, and is adapting faster, then the traditional server-setups.
